tests/cross-account-associations/main.tf

```diff
@@ -0,0 +1,195 @@
+
+  IMPORTANT: The associations in the member account are dependent on *several* resources that are not
+  direct attributes of the association. In order for terraform to properly manage the ordering of create/destroy
+  actions, use `depends_on` for the RAM share accepter, the principal association, and the resource association.
+*/
+
+provider "aws" {
+  region  = "us-east-1"
+  profile = "aws"
+}
+
+provider "aws" {
+  region  = "us-east-1"
+  alias   = "owner"
+  profile = "awsalternate"
+}
+
+module "query_log_config_association" {
+  source = "../../modules/query-log-config-association"
+
+  resolver_query_log_config_id = module.resolver.query_log_configs["${local.name}-ql1"].query_log_config.id
+  resource_id                  = module.vpc_member.vpc_id
+
+  depends_on = [
+    module.ram_share_accepter,
+    module.ram_share,
+  ]
+}
+
+module "rule_association" {
+  source = "../../modules/rule-association"
+
+  name             = local.name
+  resolver_rule_id = module.resolver.rules["${local.name}-r1"].resolver_rule.id
+  vpc_id           = module.vpc_member.vpc_id
+
+  depends_on = [
+    module.ram_share_accepter,
+    module.ram_share,
+  ]
+}
+
+module "resolver" {
+  source = "../.."
+  providers = {
+    aws = aws.owner
+  }
+
+  name               = local.name
+  direction          = "OUTBOUND"
+  security_group_ids = [aws_security_group.resolver.id]
+  tags               = {}
+  query_log_configs  = local.query_log_configs
+  rules              = local.rules
+
+  ip_addresses = [for subnet in module.vpc_owner.private_subnets : {
+    subnet_id = subnet
+    ip        = null
+  }]
+}
+
+locals {
+  id   = data.terraform_remote_state.prereq.outputs.test_id.result
+  name = "tardigrade-resolver-${local.id}"
+
+  query_log_configs = [
+    {
+      name            = "${local.name}-ql1"
+      destination_arn = aws_s3_bucket.ql1.arn
+      tags            = {}
+      associations    = []
+    },
+  ]
+
+  rules = [
+    {
+      name        = "${local.name}-r1"
+      domain_name = "foo.local"
+      rule_type   = "FORWARD"
+      tags        = {}
+      target_ips = [
+        {
+          ip   = "10.0.0.44"
+          port = null
+        },
+      ]
+      associations = []
+    },
+  ]
+}
+
+module "ram_share" {
+  source = "git::https://github.com/plus3it/terraform-aws-tardigrade-ram-share.git?ref=3.0.1"
+  providers = {
+    aws = aws.owner
+  }
+
+  name                      = "${local.name}-ram-share"
+  allow_external_principals = true
+
+  resources = [
+    {
+      name         = "${local.name}-ql1"
+      resource_arn = module.resolver.query_log_configs["${local.name}-ql1"].query_log_config.arn
+    },
+    {
+      name         = "${local.name}-r1"
+      resource_arn = module.resolver.rules["${local.name}-r1"].resolver_rule.arn
+    },
+  ]
+}
+
+module "ram_share_accepter" {
+  source = "git::https://github.com/plus3it/terraform-aws-tardigrade-ram-share.git//modules/cross_account_principal_association?ref=3.0.1"
```

Comment:
Question.  Since this module terraform-aws-tardigrade-ram-share is being used here, should the provider be set in that module?  Based on the guidance from TF "A module intended to be called by one or more other modules must not contain any provider blocks. "  I know that is separate from this PR but when doing a terraform init a warning was given

Reply:
That warning is because `terraform-aws-tardigrade-ram-share` is using some old syntax for _its_ "alias" provider block. An "alias" provider block is considered "empty" even though it contains the `alias` attribute. That's no longer necessary, but switching to the new method is backwards-incompatible with prior terraform versions, so we haven't _leapt_ to the new approach using `configuration_aliases` everywhere yet...

We can't avoid the warning in this project, no matter what we do. We will eventually need to update `terraform-aws-tardigrade-ram-share` and bump its major version.